synapse/storage/monthly_active_users.py

```diff
@@ -177,19 +177,36 @@ def user_last_seen_monthly_active(self, user_id):
             Arguments:
                 user_id (str): user to add/update
             Return:
-                Deferred[int] : timestamp since last seen, None if never seen
-
+                Deferred[(int,bool)|None] : (time since last seen, bool if trial user)
+                    None if never seen
         """
 
-        return(self._simple_select_one_onecol(
-            table="monthly_active_users",
-            keyvalues={
-                "user_id": user_id,
-            },
-            retcol="timestamp",
-            allow_none=True,
-            desc="user_last_seen_monthly_active",
-        ))
+        mau_trial_ms = self.hs.config.mau_trial_days * 24 * 60 * 60 * 1000
+
+        def _user_last_seen_monthly_active(txn):
+
+            sql = """
+                SELECT timestamp, creation_ts
+                FROM monthly_active_users LEFT JOIN users
+                ON monthly_active_users.user_id = users.name
+                WHERE user_id = ?
+            """
+
+            txn.execute(sql, (user_id,))
+            rows = txn.fetchall()
+            if not rows or not rows[0][0]:
+                return None
+            else:
+                (timestamp, created) = (rows[0][0], rows[0][1])
+                if not created:
+                    created = 0
+                trial = (timestamp - created * 1000) < mau_trial_ms
+                return (timestamp, trial)
```

Comment:
Hang on, trial users aren't added to the mau table and so they'll never be picked up by the select here?

Reply:
oh, crap, yes; my bad. should just be looking at the agr of the user in that instance and ignoring the MAU table entirely, which simplifies things even more and gets rid of the fugly left join

Reply:
oh, crap, yes; my bad. should just be looking at the agr of the user in that instance and ignoring the MAU table entirely, which simplifies things even more and gets rid of the fugly left join